File: src/main/java/org/railwaystations/rsapi/adapter/in/task/CleanupPhotoStorageTask.java
package org.railwaystations.rsapi.adapter.in.task;

import lombok.RequiredArgsConstructor;
import org.railwaystations.rsapi.core.ports.out.PhotoStorage;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class CleanupPhotoStorageTask {

    private final PhotoStorage photoStorage;

    @Scheduled(cron = "0 0 2 * * *")
    public void notifyUsers() {
        photoStorage.cleanupOldCopies();
    }

}

File: src/main/kotlin/org/railwaystations/rsapi/adapter/task/CleanupPhotoStorageTask.kt
package org.railwaystations.rsapi.adapter.task

import org.railwaystations.rsapi.core.ports.PhotoStorage
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class CleanupPhotoStorageTask(private val photoStorage: PhotoStorage) {

    @Scheduled(cron = "0 0 2 * * *")
    fun notifyUsers() {
        photoStorage.cleanupOldCopies()
    }
}
